controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -158,6 +181,40 @@ type RollingUpdate struct {
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
 
+// RemediationStrategy allows to define how control plane machine remediation happens.
+type RemediationStrategy struct {
+	// MaxRetry is the Max number of retries while attempting to remediate an unhealthy machine.
+	// A retry happens when a machine that was created as a replacement for an unhealthy machine also fails.
+	// For example, given a control plane with three machines M1, M2, M3:
+	//
+	//	M1 become unhealthy; remediation happens, and M1-1 is created as a replacement.
+	//	If M1-1 (replacement of M1) has problems while bootstrapping it will become unhealthy, and then be
+	//	remediated; such operation is considered a retry, remediation-retry #1.
+	//	If M1-2 (replacement of M1-2) becomes unhealthy, remediation-retry #2 will happen, etc.
+	//
+	// A retry could happen only after RetryPeriod from the previous retry.
+	// If a machine is marked as unhealthy after MinHealthyPeriod from the previous remediation expired,
+	// this is not considered a retry anymore because the new issue is assumed unrelated from the previous one.
+	//
+	// If not set, the remedation will be retried infinitely.
+	// +optional
+	MaxRetry *int32 `json:"maxRetry,omitempty"`
+
+	// RetryPeriod is the duration that KCP should wait before remediating a machine being created as a replacement
+	// for an unhealthy machine (a retry).
+	//
+	// If not set, a retry will happen immediately.
```

Comment:
Could we have a minimum default of 1 minute?

Reply:
If we add a defaulting rule, then it will be impossible for the users to ask for "retry as soon as possible/immediately", which I consider a valid use case. 

Also, retrying immediately is the current behavior and AFAIK no one complained about it, so I do not think it is necessary to force the system to always wait at least for some time.

Reply:
We could default to e.g. 1 minute and users could still set 0 if they want to. 
Probably have to make this a pointer ot make this possible though

Reply:
Despite what is technically feasible, let's use current behavior as a default since we don't have demand or a set of trusted data to use for picking up other values.